core/chains/evm/config/toml/defaults.go

```diff
@@ -98,39 +117,60 @@ func init() {
 			continue
 		}
 
+		// read the file to bytes
 		path := evmDir + "/" + entry.Name()
-		file, err := os.Open(path)
+		chainID, chain, err := readConfig(path, os.ReadFile, entry.Name() == "fallback.toml")
 		if err != nil {
-			log.Fatalf("error opening file (name: %v) in custom defaults override directory: %v", entry.Name(), err)
+			if errors.Is(err, errFallbackConfig) {
+				fallback = chain
+
+				continue
+			}
+
+			log.Fatalf("custom defaults override failure (%s): %s", entry.Name(), err.Error())
 		}
 
-		// Read file contents
-		b, err := io.ReadAll(file)
-		file.Close()
-		if err != nil {
-			log.Fatalf("error reading file (name: %v) contents in custom defaults override directory: %v", entry.Name(), err)
+		// ChainID as a default should not be duplicated
+		id := chainID.String()
+		if _, ok := customDefaults[id]; ok {
+			log.Fatalf("%q contains duplicate ChainID: %s", path, id)
 		}
 
-		var config = struct {
-			ChainID *big.Big
-			Chain
-		}{}
+		// set default lookups
+		customDefaults[id] = chain
+	}
+}
 
-		if err := cconfig.DecodeTOML(bytes.NewReader(b), &config); err != nil {
-			log.Fatalf("failed to decode %q in custom defaults override directory: %v", path, err)
-		}
+func readConfig(path string, reader func(name string) ([]byte, error), fallback bool) (*big.Big, Chain, error) {
+	bts, err := reader(path)
+	if err != nil {
+		return nil, Chain{}, fmt.Errorf("%w: %s", errRead, err.Error())
+	}
 
-		if config.ChainID == nil {
-			log.Fatalf("missing ChainID in: %s in custom defaults override directory. exiting", path)
-		}
+	var config = struct {
+		ChainID *big.Big
+		Chain
+	}{}
 
-		id := config.ChainID.String()
+	// decode from toml to a chain config
+	if err := cconfig.DecodeTOML(bytes.NewReader(bts), &config); err != nil {
+		return nil, Chain{}, fmt.Errorf("%w %s: %s", errDecode, path, err.Error())
+	}
 
-		if _, ok := customDefaults[id]; ok {
-			log.Fatalf("%q contains duplicate ChainID: %s", path, id)
+	if fallback {
+		if config.ChainID != nil {
+			return nil, Chain{}, fmt.Errorf("%w: found: %s", errNonNilChainID, config.ChainID)
 		}
-		customDefaults[id] = config.Chain
+
+		return nil, config.Chain, errFallbackConfig
```

Comment:
We should not be using a custom error to communicate back to the caller when there is not an error, and it seems unnecessary in this case since they passed in to us in the first place.